runtime/sema/account.cdc

```diff
@@ -0,0 +1,521 @@
+        access(Capabilities | StorageCapabilities | GetStorageCapabilityController)
+        fun forEachController(
+            forPath: StoragePath,
+            _ function: fun(&StorageCapabilityController): Bool
+        )
+
+        /// Issue/create a new storage capability.
+        access(Capabilities | StorageCapabilities | IssueStorageCapabilityController)
+        fun issue<T: &Any>(_ path: StoragePath): Capability<T>
+    }
+
+    access(all)
+    struct AccountCapabilities {
+        /// Get capability controller for capability with the specified ID.
+        ///
+        /// Returns nil if the ID does not reference an existing account capability.
+        access(Capabilities | AccountCapabilities | GetAccountCapabilityController)
+        view fun getController(byCapabilityID: UInt64): &AccountCapabilityController?
+
+        /// Get all capability controllers for all account capabilities.
+        access(Capabilities | AccountCapabilities | GetAccountCapabilityController)
+        view fun getControllers(): [&AccountCapabilityController]
+
+        /// Iterate over all account capability controllers for all account capabilities,
+        /// passing a reference to each controller to the provided callback function.
+        ///
+        /// Iteration is stopped early if the callback function returns `false`.
+        ///
+        /// If a new account capability controller is issued for the account,
+        /// or an existing account capability controller for the account is deleted,
+        /// then the callback must stop iteration by returning false.
+        /// Otherwise, iteration aborts.
+        access(Capabilities | AccountCapabilities | GetAccountCapabilityController)
+        fun forEachController(_ function: fun(&AccountCapabilityController): Bool)
+
+        /// Issue/create a new account capability.
+        access(Capabilities | AccountCapabilities | IssueAccountCapabilityController)
+        fun issue<T: &Account>(): Capability<T>
+    }
+}
+
+/* Storage entitlements */
+
+entitlement Storage
+
+entitlement SaveValue
+entitlement LoadValue
+entitlement BorrowValue
+
+/* Contract entitlements */
+
+entitlement Contracts
+
+entitlement AddContract
+entitlement UpdateContract
+entitlement RemoveContract
+
+/* Key entitlements */
+
+entitlement Keys
+
+entitlement AddKey
+entitlement RevokeKey
+
+/* Inbox entitlements */
+
+entitlement Inbox
+
+entitlement PublishInboxCapability
+entitlement UnpublishInboxCapability
+entitlement ClaimInboxCapability
+
+/* Capability entitlements */
+
+entitlement Capabilities
+
+entitlement StorageCapabilities
+entitlement AccountCapabilities
+
+entitlement PublishCapability
+entitlement UnpublishCapability
+
+entitlement GetStorageCapabilityController
+entitlement IssueStorageCapabilityController
+
+entitlement GetAccountCapabilityController
+entitlement IssueAccountCapabilityController
+
+/* Entitlement mappings */
+
+entitlement mapping AccountMapping {
+    // TODO: include Identity
```

Comment:
Will update https://github.com/onflow/cadence/issues/2643 to include this

Reply:
Will add this in a follow-up PR, as it requires additional work in the type code generator